wavefront/wavefront.go

```diff
@@ -48,25 +52,74 @@ type (
 
 	// WavefrontAdapter supports metric template.
 	WavefrontAdapter struct {
-		listener            net.Listener
-		server              *grpc.Server
-		reporterInitialized bool
+		listener net.Listener
+		server   *grpc.Server
+		reporter wf.WavefrontMetricsReporter
 	}
 )
 
 // ensure that WavefrontAdapter implements the HandleMetricServiceServer interface.
 var _ metric.HandleMetricServiceServer = &WavefrontAdapter{}
 
 // createWavefrontReporter creates a reporter that periodically flushes metrics to Wavefront.
-func createWavefrontReporter(cfg *config.Params) {
-	hostTags := map[string]string{"source": cfg.Source}
+func (wa *WavefrontAdapter) createWavefrontReporter(cfg *config.Params) {
+
 	if direct := cfg.GetDirect(); direct != nil {
-		go wf.WavefrontDirect(metrics.DefaultRegistry, cfg.FlushInterval, hostTags, cfg.Prefix, direct.Server, direct.Token)
+		directCfg := &senders.DirectConfiguration{
+			Server:               direct.Server,
+			Token:                direct.Token,
+			FlushIntervalSeconds: int(cfg.FlushInterval.Seconds()),
+			BatchSize:            10000,
+			MaxBufferSize:        50000,
+		}
+		sender, err := senders.NewDirectSender(directCfg)
+		if err != nil {
+			panic(err)
```

Comment:
Let's use `logger.Fatalf` instead of just panicking. Here and elsewhere in this file.

Reply:
Modified to use log.Fatalf instead of panic.